hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/MergeIntoHoodieTableCommand.scala

```diff
@@ -490,12 +498,32 @@ case class MergeIntoHoodieTableCommand(mergeInto: MergeIntoTable) extends Hoodie
           }
       }
     }.toSet
+  }
+
+  /**
+   * @param updatedFieldSet Updated fields based on the conditional assignments in the MERGE INTO statement.
+   * @return {@code true} if the updated fields covers all fields in the table schema;
+   *         {@code false} otherwise.
+   */
+  private def areAllFieldsUpdated(updatedFieldSet: Set[Attribute]): Boolean = {
+    mergeInto.targetTable.output
+      .filterNot(attr => isMetaField(attr.name))
+      .filter { tableAttr =>
+        !updatedFieldSet.exists(attr => attributeEquals(attr, tableAttr))
+      }
```

Comment:
Maybe `exists` is more effieicnt than `filter().isEmpty` ?

Reply:
Makes sense.  Fixed.